### PR TITLE
Add third party vector source example

### PR DESCRIPTION
**Pull request by tobrun**
Example conform to https://docs.mapbox.com/mapbox-gl-js/example/third-party/

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/style/ThirdPartyVectorSourceActivity.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    mapboxMap.setCamera(CameraOptions.Builder()
+      .zoom(12.0)
+      .center(Point.fromLngLat(-87.622088, 41.878781))
+      .build()
+    )
+
+    mapboxMap.loadStyleUri(Style.LIGHT) {
+      it.addSource(
+        vectorSource(SOURCE_ID) {
+          tiles(listOf("https://tiles.mapillary.com/maps/vtp/mly1_public/2/{z}/{x}/{y}?access_token=MLY|4142433049200173|72206abe5035850d6743b23a49c41333"))
```
Comment (alexander-kulikovskii) [inline app/src/main/java/com/mapbox/maps/testapp/examples/style/ThirdPartyVectorSourceActivity.kt:52]: should we leave it here?

File: app/src/main/res/values/example_descriptions.xml [hunk 2]
```diff
@@ -93,4 +93,5 @@
     <string name="description_info_window">Legacy InfoWindow implementation using view annotations</string>
     <string name="description_viewport">Viewport camera showcase</string>
     <string name="description_advanced_viewport">Advanced viewport with gestures showcase</string>
+    <string name="description_third_party_source">Add a third party vector tile source.</string>
```
Comment (alexander-kulikovskii) [inline app/src/main/res/values/example_descriptions.xml:96]:
```suggestion
    <string name="description_third_party_source">Third party vector tile source</string>
```

**Review by alexander-kulikovskii — APPROVED** [review]

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 3]
```diff
@@ -1393,7 +1393,7 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
     // Prevent drag start in area around horizon to avoid sharp map movements
     val topMapMargin = 0.04 * mapTransformDelegate.getSize().height
     val reprojectErrorMargin = min(10.0, topMapMargin / 2)
-    val point = ScreenCoordinate(pixel.x, pixel.y - topMapMargin)
+    val point = ScreenCoordinate(pixel.x, pixel.y - topMapMargin).sanitizeNaN()
```
Comment (yunikkk) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1396]: Suppose this is from commit from different PR

**Review by yunikkk — COMMENTED** [review]
Remove extra commit and LGTM

**Review by yunikkk — APPROVED** [review]

**Review by pengdev — APPROVED** [review]

**Pull request merged**